src/data/store/WfsFeatures.js

```diff
@@ -0,0 +1,213 @@
+     *
+     * @param {Object} config The configuration object.
+     * @private
+     */
+    constructor: function(config) {
+        var me = this;
+
+        config = config || {};
+
+        // apply count as store's pageSize
+        config.pageSize = config.count || me.count;
+
+        if (config.pageSize > 0) {
+            // calculate initial page
+            var startIndex = config.startIndex || me.startIndex;
+            var currentPage = Math.floor(startIndex / config.pageSize) + 1;
+            config.currentPage = currentPage;
+        }
+
+        // avoid creation of vector layer by parent class (raises error when
+        // applying WFS data) so we can create the WFS vector layer on our own
+        // (if needed)
+        var createLayer = config.createLayer;
+        config.createLayer = false;
+
+        me.callParent([config]);
+
+        if (!me.url) {
+            Ext.raise('No URL given to WfsFeaturesStore');
+        }
+
+        if (createLayer) {
+            // the WFS vector layer showing the WFS features on the map
+            me.source = new ol.source.Vector({
+                features: []
+            });
+            me.layer = new ol.layer.Vector({
+                source: me.source,
+                style: me.style
+            });
+
+            me.layerCreated = true;
+        }
+
+        // initally load the WFS data
+        me.loadWfs();
+
+        // before the store gets re-loaded (e.g. by a paging toolbar) we trigger
+        // the re-loading of the WFS, so the data keeps in sync
+        me.on('beforeload', me.loadWfs, me);
+
+        // add layer to connected map, if available
+        if (me.map && me.layer) {
+            me.map.addLayer(me.layer);
+        }
+    },
+
+    /**
+     * Loads the data from the connected WFS.
+     * @private
+     */
+    loadWfs: function() {
+        var me = this;
+        var url = me.url;
+        var params = {
+            service: me.service,
+            version: me.version,
+            request: me.request,
+            typeName: me.typeName,
+            outputFormat: me.outputFormat,
+            sortBy: me.sortBy
+        };
+
+        // apply paging parameters if necessary
+        if (me.pageSize) {
+            var fromRecord =
+              ((me.currentPage - 1) * me.pageSize) + me.startIndexOffset;
+            me.startIndex = fromRecord;
+            params.startIndex = me.startIndex;
+            params.count = me.pageSize;
+        }
+
+        // fire event 'gx-wfsstoreload-beforeload' and skip loading if listener
+        // function returns false
+        if (me.fireEvent('gx-wfsstoreload-beforeload', me, params) === false) {
+            return;
+        }
+
+        // request features from WFS
+        Ext.Ajax.request({
+            url: url,
+            method: 'GET',
+            params: params,
+            success: function(response) {
+                var respJson = Ext.decode(response.responseText);
```

Comment:
Does this assume JSON? I am totally fine with that, but I think we should have it in the API (or maybe I overread it)

Reply:
Please wrap `Ext.decode` in an try catch block, as it will throw when the response is in XML (which happens for invalid request / server problems)